components/api/db/cellery-hub-db.bal

```diff
@@ -72,4 +73,41 @@ public function getImageVersions(string imageId, int offset, int resultLimit) re
     return res;
 }
 
+public function getPublicArtifact(string orgName, string imageName, string artifactVersion) returns json | error {
+    log:printDebug(io:sprintf("Performing data retrieval for articat \'%s/%s:%s\'", orgName, imageName, artifactVersion));
+    table<record {}> res = check connection->select(GET_ARTIFACT_FROM_IMG_NAME_N_VERSION, gen:ArtifactResponse, orgName, imageName,
+                                                    artifactVersion, loadToMemory = true);
+    return buildJsonPayloadForGetArtifact(res, orgName, imageName, artifactVersion);
+}
+
+public function getUserArtifact(string userId, string orgName, string imageName, string artifactVersion) returns json | error {
+    log:printDebug(io:sprintf("Performing data retrieval for articat \'%s/%s:%s\'", orgName, imageName, artifactVersion));
+    table<record {}> res = check connection->select(GET_ARTIFACT_FOR_USER_FROM_IMG_NAME_N_VERSION, gen:ArtifactResponse, orgName, imageName,
+                                                    artifactVersion, userId, orgName, imageName, artifactVersion, loadToMemory = true);
+    return buildJsonPayloadForGetArtifact(res, orgName, imageName, artifactVersion);
+}
 
+function buildJsonPayloadForGetArtifact(table<record {}> res, string orgName, string imageName, string artifactVersion) returns json | error {
```

Comment:
This method should not lie within the db module should it ? Since this builds a response